packages/camera/camera/example/ios/RunnerTests/StreamingTest.m

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera;
+@import camera.Test;
+@import XCTest;
+@import AVFoundation;
+#import <OCMock/OCMock.h>
+#import "CameraTestUtils.h"
+
+@interface StreamingTests : XCTestCase
+@property(readonly, nonatomic) FLTCam *camera;
+@property(readonly, nonatomic) CMSampleBufferRef sampleBuffer;
+@end
+
+@implementation StreamingTests
+
+- (void)setUp {
+  dispatch_queue_t captureSessionQueue = dispatch_queue_create("testing", NULL);
+  _camera = FLTCreateCamWithCaptureSessionQueue(captureSessionQueue);
+  _sampleBuffer = FLTCreateTestSampleBuffer();
+}
+
+- (void)tearDown {
+  CFRelease(_sampleBuffer);
+}
+
+- (void)waitForStart {
+  NSPredicate *predicate = [NSPredicate predicateWithFormat:@"isStreamingImages == YES"];
+  XCTNSPredicateExpectation *expectation =
+      [[XCTNSPredicateExpectation alloc] initWithPredicate:predicate object:_camera];
+  XCTWaiterResult result = [XCTWaiter waitForExpectations:@[ expectation ] timeout:3];
```

Comment:
i'm curious will it fail if we use 1 second? due to predicate parsing? if so please document it out. otherwise let's keep 1 second.

Reply:
also i think you can do `[self expectationForPredicate...]` and `[self waitForExpectations:]`

Reply:
Using XCTNSPredicateExpectation, it seems to time out with a 1 second wait.
I tried using KVO this time and it seems to work well.